patches/net/minecraft/server/level/ServerPlayerGameMode.java.patch

```diff
@@ -60,16 +60,18 @@
 -                    boolean flag = this.player.hasCorrectToolForDrops(blockstate);
 +                    boolean flag1 = blockstate.canHarvestBlock(this.level, p_9281_, this.player); // previously player.hasCorrectToolForDrops(blockstate)
                      itemstack.mineBlock(this.level, blockstate, p_9281_, this.player);
+-                    if (flag1 && flag) {
+-                        block.playerDestroy(this.level, this.player, p_9281_, blockstate, blockentity, itemstack1);
 +                    if (itemstack.isEmpty() && !itemstack1.isEmpty())
 +                        net.neoforged.neoforge.event.EventHooks.onPlayerDestroyItem(this.player, itemstack1, InteractionHand.MAIN_HAND);
-+                    boolean flag = removeBlock(p_9281_, blockstate, flag1);
++                    BlockState blockState = removeBlock(p_9281_, blockstate, flag1);
```

Comment:
Use a different name here.  We don't want to have a local that is the same name as another local but with different capitalization.